backend/app/anti_recommenders/anti_recommender.py

```diff
@@ -11,6 +11,6 @@ class AntiRecommender(ABC):
 
     @abstractmethod
     def generate_anti_recommendations(
-        self, record_key: str
-    ) -> Generator[AntiRecommendation, None, None]:
+        self, record_key: str, record_type: str
```

Comment:
Nit: it feels like there should be `self, *, record_key: str, record_type: str` so you have to keyword-specify parameters. I usually do that with > 1 parameter, and always if 2+ parameters have the same type. Another way you could deal with this is tiny types for `record_key` and/or `record_type`.